SQLInjectionFlaskApp/app.py

```diff
@@ -23,32 +23,46 @@ def success(name):
 def search_inventory():
     return render_template('search_inventory.html', search_result=[{}])
 
+def getSearchResults(item):
+    try:
+        connection = connect(host='',
+                                user='root',
+                                password='mysqlroot',
+                                database='inventory',
+                                cursorclass=cursors.DictCursor)
+        with connection:
+            with connection.cursor() as cursor:
+                # select * from items where name like '%hammer' UNION (SELECT TABLE_NAME, TABLE_SCHEMA FROM information_schema.tables);-- %'
+                # hammer' UNION (SELECT TABLE_NAME, TABLE_SCHEMA FROM information_schema.tables);--
+                sql_command = "select * from items where item_name like '%%"+item+"%'"
+                cursor.execute(sql_command)
+                result = cursor.fetchall()
+                if not result: result = [{}]
+        return { "success": True, "result": result }
+    except Exception as e:
+        return { "success": False, "error": e }
 
 @app.route('/submit', methods=['POST', 'GET'])
 def submit():
-    result = ''
     item = request.form['item']
     if request.method == 'POST':
-        try:
-            connection = connect(host='',
-                                 user='root',
-                                 password='mysqlroot',
-                                 database='inventory',
-                                 cursorclass=cursors.DictCursor)
-            with connection:
-                with connection.cursor() as cursor:
-                    # select * from items where name like '%hammer' UNION (SELECT TABLE_NAME, TABLE_SCHEMA FROM information_schema.tables);-- %'
-                    # hammer' UNION (SELECT TABLE_NAME, TABLE_SCHEMA FROM information_schema.tables);--
-                    sql_command = "select * from items where item_name like '%%"+item+"%'"
-                    cursor.execute(sql_command)
-                    result = cursor.fetchall()
-        except Exception as e:
-            return render_template('search_inventory.html', search_result=[{"error": str(e)}])
-
-        return render_template('search_inventory.html', search_result=result)
+        searchRes = getSearchResults(item)
+        if searchRes["success"]:
+            return render_template('search_inventory.html', search_result=searchRes["result"])
+        else:
+            return render_template('search_inventory.html', search_result=[{"error": str(searchRes["error"])}])
     else:
         return render_template('search_inventory.html', search_result=[{}])
 
+@app.route('/status', methods=['GET'])
+def getProblemStatus():
+    status = { "solved": False }
+    item = "hammer' UNION (SELECT TABLE_NAME, TABLE_SCHEMA FROM information_schema.tables);-- "
+    searchRes = getSearchResults(item)
+    if not searchRes["success"] or not searchRes["result"]: return status
```

Comment:
The `searchRes["success"]` is always `True` both in case of vulnerable as well as safe versions. I think this check is not needed. Only time `searchRes["success"]` is `False` is when the SQL command is wrong and this appears as an error in the UI.